pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -498,6 +503,38 @@ func (a *Actuator) ensureAnnotation(ctx context.Context, machine *machinev1beta1
 	return true, a.client.Update(ctx, machine)
 }
 
+func (a *Actuator) ensureProviderID(ctx context.Context, machine *machinev1beta1.Machine, host *bmh.BareMetalHost) (error) {
```

Comment:
The parenthesis around `error` are not necessary, but not harmful. Though I suspect `go fmt` would remove them. Probably best to go ahead and remove them here.

Reply:
Also a doc string is not strictly required for a non-exported function, but it would be useful and consistent with the rest of the code in this package.

Reply:
Now added

Reply:
Just making sure the comment wasn't lost: thoughts on the parens around `error`? Quick testing shows `go fmt` will remove them, and it's the authority on style.

Reply:
Ah yes sorry I did miss that, now updated after running `make fmt` which also fixed some other cosmetic issues.